scripts/src/adaptTranslationCsv.ts

```diff
@@ -0,0 +1,70 @@
+import csvParse = require("csv-parse/lib/sync");
+import { readFileSync, writeFile } from "fs";
+import { path } from "./utils";
+import { CsvParse, TranslationData } from "./admin/translations";
+import { ExportToCsv, Options as ExportCsvOptions } from "export-to-csv";
+
+function getOfferTranslation(inputCsv: TranslationData) {
+	return inputCsv.reduce((dict, [type, key, defaultStr, translation]) => {
+		if (!translation || !translation.length) {
+			return dict;
+		}
+		dict[defaultStr] = translation;
+		return dict;
+	}, {} as { [defaultStr: string]: string });
+
+}
+
+async function addTranslationTo(csv: TranslationData, fromDict: { [defaultStr: string]: string }) {
+	return csv.map(([type, key, defaultStr, __, charLimit]) => {
+		return {
+			Type: type,
+			Key: key,
+			Default: defaultStr,
+			Translation: fromDict[defaultStr] || defaultStr,
+			"Character Limit": charLimit,
+		};
+	});
+}
+
+function writeCsvDataToFile(data: any[], fileName: string) {
+	const options: ExportCsvOptions = {
+		fieldSeparator: ",",
+		quoteStrings: "\"",
+		decimalseparator: ".",
+		showLabels: true,
+		showTitle: false,
+		useBom: true,
+		useKeysAsHeaders: true,
+	};
+
+	const csvExporter = new ExportToCsv(options);
+	writeFile(fileName, csvExporter.generateCsv(data, true), (err: NodeJS.ErrnoException) => {
```

Comment:
here also i don't understand why the type of the error is important, it can just be `err => { ...`